src/plugins/vis_types/xy/public/utils/accessors.tsx

```diff
@@ -79,8 +79,16 @@ export const getSplitSeriesAccessorFnMap = (
 };
 
 // For percentile, the aggregation id is coming in the form %s.%d, where %s is agg_id and %d - percents
-export const getSafeId = (columnId?: number | string | null) =>
-  (columnId || '').toString().split('.')[0];
+export const getSafeId = (columnId?: number | string | null) => {
+  const id = String(columnId);
+  // only multi-value aggs like percentiles are allowed to contain dots and [
+  const isMultiValueId = id.includes('[') || id.includes('.');
+  if (!isMultiValueId) {
+    return id;
+  }
+  const baseId = id.substring(0, id.indexOf('[') !== -1 ? id.indexOf('[') : id.indexOf('.'));
```

Comment:
nit: either one of the two must be positive:
```suggestion
  const baseId = id.substring(0, Math.max(id.indexOf('['), id.indexOf('.')));
```